### ISSUE TITLE
Can I use TDD frameworks like minitest to test instances, not classes?

### ISSUE THREAD

**Issue by datenimperator** [seattlerb/minitest#857]
**Prelude:** _There's stuff going on on my network._ Things like web server configs and DNS names and SSL certs and such need to be tested regularly, and I'm about to write something in Ruby in order to do so. I'll have classes defining test methods, I'll need runners and loggers, not to mention basic benchmarking and such. So I wonder: **Can I use my TDD tool minitest for that?** Which already has all that and more?

Problem here: I'm not going to test classes, but instead instances of such classes. Usually there's a class to be tested, and another class implementing such tests as instance methods. The runner's responsible for creating instances of the subject and test. Here, I'd need to perform a single test class multiple times, against different target instances. Is that possible?

**Comment by zenspider** [comment]
At the end of the day... minitest just runs methods. It doesn't care what you do in them... so absolutely.

**Comment by datenimperator** [comment]
Ryan, thx a lot. When I started to look into this, I did not manage to find a way how to do this with Minitest et al. I ended up writing some lines of code to implement things like a runner and some "assert" methods. It turned out to be a JSON-based declaration of things to test, plus some Ruby code to perform these tests and report them in a junit-y way so my CI can consume the results. Meet "Sanity": https://git.software-berater.net/christian/sanity
